backend/windmill-worker/src/worker.rs

```diff
@@ -1746,6 +1691,317 @@ pub async fn run_worker<R: rsmq_async::RsmqConnection + Send + Sync + Clone + 's
+            SpawnWorker::Script { path, .. } => path.to_string(),
+        };
+        let path2 = path.clone();
+        let w_id = w_id.to_string();
+
+        let (content, lock, language, envs) = match sw {
+            SpawnWorker::Script { path, hash } => {
+                let r;
+                loop {
+                    let q = if let Some(hash) = hash {
+                        get_script_content_by_hash(&hash, &w_id, &db).await.map(
+                            |r: ContentReqLangEnvs| {
+                                Some((r.content, r.lockfile, r.language, r.envs))
+                            },
+                        )
+                    } else {
+                        sqlx::query_as::<_, (String, Option<String>, Option<ScriptLang>, Option<Vec<String>>)>(
+                        "SELECT content, lock, language, envs FROM script WHERE path = $1 AND workspace_id = $2 AND
+                            created_at = (SELECT max(created_at) FROM script WHERE path = $1 AND workspace_id = $2 AND
+                            deleted = false AND lock IS not NULL AND lock_error_logs IS NULL)",
+                    )
+                    .bind(&path)
+                    .bind(&w_id)
+                    .fetch_optional(&db)
+                    .await
+                    .map_err(|e| Error::InternalErr(format!("expected content and lock: {e}")))
+                    };
+
+                    if let Ok(q) = q {
+                        if let Some(wp) = q {
+                            r = wp;
+                            break;
+                        } else {
+                            tracing::error!(
+                        "Failed to fetch script `{}` in workspace {} for dedicated worker. Retrying in 10s.",
+                        path,
+                        w_id
+                    );
+                            tokio::select! {
+                                biased;
+                                _ = killpill_rx.recv() => {
+                                    tracing::info!("Killing dedicated worker while it was attempting to fetch script");
+                                    return None;
+                                }
+                                _ = tokio::time::sleep(Duration::from_secs(10)) => {
+                                    continue;
+                                }
+
+                            }
+                        }
+                    } else {
+                        tracing::error!("Failed to fetch script for dedicated worker");
+                        killpill_tx.send(()).expect("send");
+                        return None;
+                    }
+                }
+                r
+            }
+            SpawnWorker::RawScript { content, lock, lang, .. } => (content, lock, Some(lang), None),
+        };
+
+        match language {
+            Some(ScriptLang::Python3) | Some(ScriptLang::Bun) | Some(ScriptLang::Deno) => {}
+            _ => return None,
```

Comment:
I can't find just looking at the code where we handle the error if the flow contains one module in a non supported language, but I'm sure you're handling that case since we discussed it yesterday

Reply:
It's the None case here, will discuss in daily